### PR TITLE
[CIS-661] Simplify attachment send API

### PR DESCRIPTION
**Pull request by mantergo**
**In this PR:**

- Simplify attachment send API

### CODE REVIEW

**Review by b-onc — APPROVED** [review]
IMHO it looks much nicer now 🎉 
Pro: Simplified API, unified flow
Con: Docs are basically required to understand these things, but `seeds` were not self-explanatory anyway

File: Sources/StreamChat/Models/Attachments/ChatMessageAttachmentSeed.swift [hunk 1]
```diff
@@ -55,6 +55,9 @@ public struct ChatMessageAttachmentSeed: Hashable {
         self.type = type
         self.localState = localState
     }
+    
+    // Dummy encodable conformance to satisfy `AttachmentEnvelope` protocol.
+    public func encode(to encoder: Encoder) throws {}
```
Comment (VojtaStavik) [inline Sources/StreamChat/Models/Attachments/ChatMessageAttachmentSeed.swift:60]: This is an interesting side effect. I'm thinking - what does this mean? Maybe we're missing something?

**Review by VojtaStavik — APPROVED** [review]
Good idea @mantergo and @b-onc  👍

File: Sources/StreamChat/Models/Attachments/ChatMessageAttachmentSeed.swift (hunk 1 above)
Comment (mantergo) [reply Sources/StreamChat/Models/Attachments/ChatMessageAttachmentSeed.swift:60]: `Encodable` is the main constraint for the custom attachments cause it is the minimum we need to blindly send it to the backend.
Dummy conformance means that we can't blindly send it, some pre-processing is required and we know how to handle that.
I think that's acceptable trade-off.

**Pull request merged**